sources/packages/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -268,4 +212,42 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
     } as Note;
     return this.repo.save(studentRestrictionEntity);
   }
+
+  /**
+   * The service function checks if the requested student has
+   * the restricted restriction.
+   * @param studentId student Id
+   * @param restrictionActions list of restriction actions
+   * @param checkAll,is by default false, checkAll decides if the all the
+   * elements in the restrictionActions should be checked (i.e checkAll = true)
+   * or any one of the elements is to be checked (i.e checkAll = false).
+   * @returns boolean, true if the restriction action is present
+   * for the student, else false.
+   */
+  async hasRestrictionAction(
+    studentId: number,
+    restrictionActions: RestrictionActionType[],
+    checkAll = false,
+  ): Promise<boolean> {
+    const query = this.repo
```

Comment:
Is there a reason why this method cannot just call the `getExistsBlockRestrictionQuery` and do the `.getRawOne()`?

Reply:
in `getExistsBlockRestrictionQuery ` studentid is not a aparameter